src/Form/Mixin/When.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Kirby\Form\Mixin;
+
+/**
+ * @package   Kirby Form
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://opensource.org/licenses/MIT
+ */
+trait When
+{
+	protected array|null $when = null;
+
+	/**
+	 * Checks if the field is currently active
+	 * or hidden because of a `when` condition
+	 */
+	public function isActive(): bool
+	{
+		if ($this->when === null || $this->when === []) {
+			return true;
+		}
+
+		$siblings = $this->siblings();
+
+		foreach ($this->when as $field => $value) {
+			$field = $siblings->get($field);
+			$input = $field?->value() ?? '';
+
+			// if the input data doesn't match the requested `when` value,
+			// that means that this field is not required and can be saved
+			// (*all* `when` conditions must be met for this field to be required)
+			if ($input !== $value) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Checks if the field needs a value before being saved;
+	 * this is the case if all of the following requirements are met:
+	 * - The field is saveable
+	 * - The field is required
+	 * - The field is currently empty
+	 * - The field is not currently inactive because of a `when` rule
+	 */
+	protected function needsValue(): bool
+	{
+		if (
+			$this->isSaveable() === false ||
+			$this->isRequired() === false ||
+			$this->isEmpty() === false ||
+			$this->isActive() === false
+		) {
+			return false;
+		}
+
+		return true;
+	}
```

Comment:
Really unsure if this shouldn't be in the validation mixin. It's somewhat part of both.

Reply:
I'm working on additional steps with a new Value mixin. I think it should actually go in there later. Should I move it back to the classes for this PR or do you think it's fine to move it to a different Mixin later?

Reply:
It's fine to have it in the mixing and move it later.